Automattic/abacus
# Make experiment p2 url optional

Fixes https://github.com/Automattic/experimentation-platform/issues/947

### Summary:

* Updates schema to allow `p2Url` to be optional in the Experiment wizard and update labels.
* Add a field to require a valid `p2Url` when launching an experiment.
* Switch from using a `patch` request to a `put` request with existing experiment data + the two fields in the modal. 

<img width="641" alt="Screenshot 2023-05-19 at 01 01 30" src="https://github.com/Automattic/abacus/assets/14192054/acfd6b2e-8d67-425d-af72-b3ff71492ee5">

### Note:
<s>Similar to https://github.com/Automattic/abacus/pull/871, because the `PATCH` request is happening after the change of status from `staging` to `running`, there is a slight possibility that a running experiment may end up without a P2 url. We'll be adding the functionality to add/update it in the Experiment Overview tab of an experiment in https://github.com/Automattic/abacus/issues/609</s>

As mentioned in https://github.com/Automattic/abacus/pull/873#issuecomment-1555115334, we're now updating the experiment (using `put`) with the new data from the modal **before** launching.

## How has this been tested?






- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)
- Manual testing with mock data (locally or on Vercel)
- Additional manual testing instructions (please specify): 
  - Apply D111312-code
  - Create an experiment without providing a `p2 Url`
  - Launch the experiment by filling in all the required fields 
    * <s>The actual launch with currently throw an error because. See D111312-code#2221821</s> 
    Fixed by updating the experiment before launching + cleaning up an old validation check. See D111312-code#inline-377792
